dask/dataframe/rolling.py

```diff
@@ -0,0 +1,47 @@
+from functools import partial, wraps
+from itertools import count
+
+from toolz import merge
+import pandas as pd
+
+tokens = ('rolling-%d' % i for i in count(1))
+
+
+def rolling_chunk(func, part1, part2, window, *args):
+    if part1.shape[0] < window:
+        raise NotImplementedError("Window larger than partition size")
+    extra = window - 1
```

Comment:
i think this should be `window + 1` and not `window - 1`

if `extra == window - 1` then a window size of 1 will concat each chunk to the next one and then doing a rolling window on it, with `window == 0` the results should be all `nan` the size of the `series` and currently we get all `nan` truncated to the size of `npartitions + len(chunk)`. While these *are* edges case, pandas handles it correctly.